generated/nidaqmx/system/device.py

```diff
@@ -1253,6 +1313,54 @@ def self_test_device(self):
         self._interpreter.self_test_device(
             self._name)
 
+    def write_id_pin_memory(self, id_pin_name, data, format_code):
+        """
+        Writes the supplied data and format code to the EEPROM connected
+        to the specified ID pin.
+
+        Args:
+            id_pin_name (str): Is the name of the ID Pin to access (ex:
+                id0)
+            data (List[int]): Is the 1D list of 8-bit unsigned integers
+                to write to the memory
+            format_code (int): Specifies the type and structure of the
+                data being written. User to define the meaning of the
+                format code.
+        """
+        if data is None:
+            data = []
+
+        data = numpy.array(data, dtype=numpy.uint8)
+
+
+        self._interpreter.write_id_pin_memory(
+            self._name, id_pin_name, data, format_code)
+
+    # region ID Pin Hand-written Functions
+
+    def read_id_pin_memory(self, id_pin_name):
+        """
+        Reads and returns the data stored in the memory attached to the
+        specified ID Pin.
+
+        Args:
+            id_pin_name (str): Is the name of the ID Pin to access (ex:
+                id0)
+
+        Returns:
+            Tuple[numpy.ndarray, int]: Returns a tuple containing
+            the data read from the ID Pin and the format code of the
+            data read.
```

Comment:
I can't find an existing method that returns a "raw" tuple like this. We tend to use Named Tuples for a bit more type-safety. See [types.py](https://github.com/ni/nidaqmx-python/blob/601db52da9642f2a760ba331e42217d85921de10/generated/nidaqmx/types.py#L3) and [DriverVersion](https://github.com/ni/nidaqmx-python/blob/601db52da9642f2a760ba331e42217d85921de10/generated/nidaqmx/system/system.py#L73). I think that would be useful here, too. You could inline a named Tuple above this method in the mako pretty easily